File: BilisPera/BilisPera/Categories/UITableViewCell+Extension.m
//
//  UITableViewCell+Extension.m
//  BilisPera
//
//  Created by BHJ on 2025/8/12.
//

#import "UITableViewCell+Extension.h"

@implementation UITableViewCell (Extension)

- (NSString *)reuseId{
    return [NSString stringWithFormat:@"reusable_%@",[self class]];
}

@end

File: SkyLoan/SkyLoan_phillipines/Extension/UITableViewCell+Extension.swift
//
//  UITableViewCell+Extension.swift
//  SkyLoan_phillipines
//
//  Created by BHJ on 2025/5/15.
//

import UIKit


extension UITableViewCell{
    static private var dataKey = "dataKey"
    
    static var reusableId: String{
        NSStringFromClass(Self.self)
    }
    
    var data: Any?{
        set{
            if let value = newValue{
                objc_setAssociatedObject(self, &UITableViewCell.dataKey, value, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
            }
            configData(data: newValue)
        }
        get{
            objc_getAssociatedObject(self, &UITableViewCell.dataKey)
        }
    }
}
